src/libxtp/davidsonsolver.cc

```diff
@@ -235,5 +380,59 @@ Eigen::MatrixXd DavidsonSolver::gramschmidt_ortho(const Eigen::MatrixXd &A,
   return Q;
 }
 
+void DavidsonSolver::restart (const DavidsonSolver::RitzEigenPair &rep,
+   DavidsonSolver::ProjectedSpace &proj, int size_restart) const {
+  proj.V = rep.q.leftCols(size_restart);
+  proj.V.colwise().normalize();
+  proj.AV = proj.AV * rep.U.leftCols(size_restart);  // corresponds to replacing V with
+                                     // q.leftCols
+  proj.T = proj.V.transpose() * proj.AV;
+  proj.search_space = size_restart;
+}
+
+void DavidsonSolver::storeConvergedData(const DavidsonSolver::RitzEigenPair &rep, 
+    int neigen, int iiter) {
+
+  DavidsonSolver::storeEigenPairs(rep,neigen);
+  this->_num_iter = iiter;
```

Comment:
actually iiter could be replaced by _num_iter everywhere.

Reply:
I am not sure, depends on how you want it. Your programming is very functional with very little member variables, that is good in  general, but leads to many function arguments. Maybe you can pull another class out or so?

Reply:
I like it better like this. iiter is a local variable and we simply export to _num_iter at the end